tool/tctl/common/notification_command.go

```diff
@@ -0,0 +1,313 @@
+func (n *NotificationCommand) Initialize(app *kingpin.Application, _ *servicecfg.Config) {
+	notif := app.Command("notifications", "Manage cluster notifications.")
+
+	n.create = notif.Command("create", "Create a cluster notification.").Alias("add")
+	n.create.Flag("user", "Target a specific user.").StringVar(&n.user)
+	n.create.Flag("roles", "Target a specific set of roles. By default, this will target all users with any of the provided roles, use --require-all-roles to exclusively target users with all of them.").StringsVar(&n.roles)
+	n.create.Flag("require-all-roles", "Set whether this notification should target users who have all of the provided roles.").BoolVar(&n.requireAllRoles)
+	n.create.Flag("title", "Set the notification's title.").Short('t').Required().StringVar(&n.title)
+	n.create.Flag("content", "Set the notification's content.").Required().StringVar(&n.content)
+	n.create.Flag("warning", "Set whether this notification is a warning notification.").BoolVar(&n.warning)
+
+	n.ls = notif.Command("ls", "List notifications which were manually created using `tctl notifications create`. By default, this will list notifications capable of targeting multiple users, such as role-based ones. To list notifications directed only at a specific user, use the --user flag. To include notifications generated by Teleport, use --all.")
+	n.ls.Flag("user", "Set which user to list user-specific notifications for.").StringVar(&n.user)
+	n.ls.Flag("format", "Output format, 'yaml', 'json', or 'text'").Default(teleport.Text).EnumVar(&n.format, teleport.YAML, teleport.JSON, teleport.Text)
+	n.ls.Flag("all", "Set whether all notifications should be included, including those generated by Teleport, as opposed to solely those created using `tctl notifications create`.").BoolVar(&n.allNotifications)
+
+	n.rm = notif.Command("rm", "Remove a cluster notification.").Alias("remove")
+	n.rm.Flag("user", "The user the notification to remove belongs to, if any.").StringVar(&n.user)
+	n.rm.Arg("id", "The ID of the notification to remove.").Required().StringVar(&n.title)
+
+	if n.stdout == nil {
+		n.stdout = os.Stdout
+	}
+}
+
+// TryRun takes the CLI command as an argument and executes it.
+func (n *NotificationCommand) TryRun(ctx context.Context, cmd string, client *authclient.Client) (match bool, err error) {
+	nc := client.NotificationServiceClient()
+
+	switch cmd {
+	case n.create.FullCommand():
+		err = n.Create(ctx, client)
+	case n.ls.FullCommand():
+		err = n.List(ctx, nc)
+	case n.rm.FullCommand():
+		err = n.Remove(ctx, client)
+	default:
+		return false, nil
+	}
+	return true, trace.Wrap(err)
+}
+
+// Create creates a new notification.
+func (n *NotificationCommand) Create(ctx context.Context, client *authclient.Client) error {
+	meta := &headerv1.Metadata{
+		Labels: map[string]string{
+			types.NotificationTitleLabel:       n.title,
+			types.NotificationTextContentLabel: n.content,
```

Comment:
We did this so we could use the `Metadata` struct to contain all of a notification's metadata, since every unique type of notification will have completely different metadata fields. For example, this `TextContent` field exists only for user-created notifications, and no other notification type will have it. The only field that is universal among all notifications is `Title`, which could be in `Spec` but for the sake of consistency and keeping all metadata grouped together in one place we decided to store it here alongside all other notification metadata.